api/envoy/admin/v2alpha/config_dump.proto

```diff
@@ -219,3 +220,31 @@ message ScopedRoutesConfigDump {
   repeated DynamicScopedRouteConfigs dynamic_scoped_route_configs = 2
       [(gogoproto.nullable) = false];
 }
+
+// Envoys SDS implementation fills this message with all secrets fetched dynamically via SDS.
+message SecretsConfigDump {
+  // DynamicSecret contains secret information fetched via SDS.
+  // The security sensitive information will trimmed, for example, private key.
+  message DynamicSecret {
+    // The name assigned to the secret.
+    string name = 1;
+
+    // This is the per-resource version information.
+    string version_info = 2;
+
+    // The timestamp when the secret was last updated.
+    google.protobuf.Timestamp last_updated = 3;
+
+    // The actual secret information. Security sensitive information is stripped.
+    // For example, private key in a TLS cert.
+    envoy.api.v2.auth.Secret secret = 4;
+  }
+
+  // The dynamically loaded active secrets. These are secrets that are available to service
+  // clusters or listeners.
+  repeated DynamicSecret dynamic_active_secrets = 1;
+
+  // The dynamically loaded warming secrets. These are secrets that are currently undergoing
+  // warming in preparation to service clusters or listeners.
+  repeated DynamicSecret dynamic_warming_secrets = 2;
+}
```

Comment:
Envoy uses "warming" to describe listeners/clusters waiting for _another_ resource to be fetched, not on itself, and we don't distinguish this "pending" state for any other resource, as far as I can tell.

Having said that, this seems useful for debugging, so perhaps we should add it to other resources as well, potentially using different term (e.g. "pending")?

@mattklein123 @htuch any thoughts?

Reply:
No strong preference here. IMO "warming" makes sense to me as a general way of describing "waiting for initialization" so this seems fine to me.

Reply:
Haven't changed the naming yet. I'm fine with either, we can either as mentioned above, enlarge the definition of the `warming` or use new terms.

let me know if you think this better changed.